src/lib/ares_process.c

```diff
@@ -71,35 +71,66 @@ static void          end_query(ares_channel_t *channel, struct query *query,
                                ares_status_t status,
                                const ares_dns_record_t *dnsrec);
 
-static void          server_increment_failures(struct server_state *server)
+static void server_increment_failures(struct server_state *server)
 {
-  ares__slist_node_t   *node;
-  const ares_channel_t *channel = server->channel;
+  ares__slist_node_t *node;
+  ares_channel_t     *channel = server->channel;
+  char                server_ip[INET6_ADDRSTRLEN];
 
```

Comment:
Move `server_ip` under the `if()` statement where it is used.  c89 allows variable declarations at the beginning of any code block.

Reply:
I will move the variable declaration (I'm used to subtly different coding patterns and need to break the habit!).

On writing out the server, I notice that `ares_get_servers_csv()` has logic for writing out a server to a string (including port and interface). Perhaps we could pull this out into a function, and re-use it for the server state callback?

Reply:
yeah, that would make sense.  It might also be good to have an unused 'flags' field for additional information about the server we can use in the future if we want (like indicating mDNS, DNS over HTTPS, etc).

Reply:
I moved out the logic to write a sever to a buffer into a new function:
`ares_status_t ares_write_server(const struct server_state *server, ares__buf_t *buf)`

In terms of future-proofing for things like mDNS and DNS over HTTPS, I suppose that info would likely be stored on the `server_state` struct? In which case `ares_write_server()` would have the info available.